src/devices/Mlx90640/Mlx90640.cs

```diff
@@ -0,0 +1,957 @@
+                    }
+
+                    irData = irData - _tgc * irDataCP[subPage];
+                    irData = irData / emissivity;
+
+                    float alphaCompensated = (float)(SCALEALPHA * alphaScale / _alpha[pixelNumber]);
+                    alphaCompensated = (float)(alphaCompensated * (1 + _ksTa * (ta.DegreesCelsius - 25)));
+
+                    float sx = (float)(alphaCompensated * alphaCompensated * alphaCompensated * (irData + alphaCompensated * taTr.Kelvins));
+                    sx = (float)(Math.Sqrt(Math.Sqrt(sx)) * _ksTo[1]);
+
+                    float to = (float)(Math.Sqrt(Math.Sqrt(irData / (alphaCompensated * (1 - _ksTo[1] * 273.15) + sx) + taTr.Kelvins)) - 273.15);
+
+                    byte range;
+                    if (to < _ct[1])
+                    {
+                        range = 0;
+                    }
+                    else if (to < _ct[2])
+                    {
+                        range = 1;
+                    }
+                    else if (to < _ct[3])
+                    {
+                        range = 2;
+                    }
+                    else
+                    {
+                        range = 3;
+                    }
+
+                    to = (float)(Math.Sqrt(Math.Sqrt(irData / (alphaCompensated * alphaCorrR[range] * (1 + _ksTo[range] * (to - _ct[range]))) + taTr.Kelvins)) - 273.15);
+
+                    result[pixelNumber] = Temperature.FromDegreesCelsius(to);
+                }
+            }
+        }
+
+        private Temperature GetTa(ushort[] frameData)
+        {
+            float ptat = (short)frameData[800];
+            float ptatArt = (short)frameData[PixelCount];
+            ptatArt = (float)((ptat / (ptat * _alphaPTAT + ptatArt)) * Math.Pow(2, 18));
+
+            float vdd = GetVdd(frameData);
+            float ta = (float)((ptatArt / (1 + _kvPTAT * (vdd - 3.3)) - _vPTAT25));
+            ta = ta / _ktPTAT + 25;
+
+            return Temperature.FromDegreesCelsius(ta);
+        }
+
+        private float GetVdd(ushort[] frameData)
+        {
+            float vdd = (short)frameData[810];
+            int resolutionRAM = (frameData[832] & 0x0C00) >> 10;
+            float resolutionCorrection = (float)(Math.Pow(2, _resolutionEE) / Math.Pow(2, resolutionRAM));
+            vdd = (float)(((resolutionCorrection * vdd - _vdd25) / _kVdd) + 3.3);
+
+            return vdd;
+        }
+
+        private ushort GetFrameData(ref ushort[] frameData)
+        {
+            Span<ushort> statusRegister = stackalloc ushort[0];
+            do
+            {
+                statusRegister = ReadRegisters16(0x8000, 1);
+            }
+            while ((byte)(statusRegister[0] & 0x0008) == 0);
+
+            byte cnt = 0;
+            Span<ushort> framedata = stackalloc ushort[832];
+
+            do
+            {
+                Write16(0x8000, 0x0030);
```

Comment:
Would be good to have a reference in the documentation for those 2 magic numbers